packages/latte-phpstan-compiler/src/PhpParser/NodeVisitor/LinkNodeVisitor.php

```diff
@@ -125,4 +153,42 @@ private function createLinkParams(Array_ $array): array
 
         return $linkParams;
     }
+
+    private function remapTarget(string $targetName): string
+    {
+        $targetName = ltrim($targetName, '/:');
+
+        if (str_ends_with($targetName, '!')) {
+            return $targetName;
+        }
+
+        $targetNameParts = explode(':', $targetName);
+        if (count($targetNameParts) === 3) {
+            return $targetName;
+        }
+
+        if ($this->actualclass === null) {
+            return $targetName;
+        }
+
+        $presenterFactory = $this->presenterFactoryFaker->getPresenterFactory();
+        $presenterName = @$presenterFactory->unformatPresenterClass($this->actualclass->getTypeAsString());
+
+        if ($presenterName === null) {
+            return $targetName;
+        }
+
+        if (count($targetNameParts) === 1) {
+            return $presenterName . ':' . $targetName;
+        }
+
+        $presenterNameParts = explode(':', $presenterName, 2);
+
+        $module = isset($presenterNameParts[1]) ? $presenterNameParts[0] : null;
+        if ($module) {
+            return $module . ':' . $targetName;
+        }
+
+        return $targetName;
+    }
```

Comment:
This method looks pretty complex and easy to break. Please cover it with test, so can change it safely

Reply:
Thanks for sharing. Now this know-how is PR-locked and will not be part of code-base. Any change of unused method will pass, but will be reported as bug too late. 

These must be in test, so it's always part of the code as verification and safety net.

Reply:
I've added LinkDestinationProcessor instead of remapTarget method and cover it with tests